operators/src/plot/box_plot.rs

```diff
@@ -293,18 +281,10 @@ impl BoxPlotRasterQueryProcessor {
                 let tile = tile?;
 
                 match tile.grid_array {
-                    GridOrEmpty::Empty(grid) if include_no_data => {
-                        let v:f64 = grid.no_data_value().expect("Empty grids always have a no-data value").as_();
-                        let iter = std::iter::repeat(v).take(grid.number_of_elements());
-                        accum.update(iter)?;
-                    },
                     // Ignore empty grids if no_data should not be included
                     GridOrEmpty::Empty(_) => {},
-                    GridOrEmpty::Grid(grid) if include_no_data => {
-                        accum.update(grid.data.iter().map(|x| (*x).as_()))?;
-                    },
                     GridOrEmpty::Grid(grid) => {
-                        accum.update(grid.data.iter().filter(|&x| !grid.is_no_data(*x)).map(|x| (*x).as_()))?;
+                        accum.update(grid.masked_copy_element_iterator().filter_map(|pixel_option| pixel_option.map(|p| { let v: f64 = p.as_(); v})))?; // TODO: if empty pixels shuld be included we need to add an enmpty pixel count in the accu
```

Comment:
I guess this was the include_nodata flag? But where would that be drawn in a boxplot?

Also: why do the elements have to be copied here?